src/ux/patchesView.ts

```diff
@@ -140,201 +138,96 @@ export const patchesTreeDataProvider: TreeDataProvider<
-                    : undefined,
-                label: filename,
-                description: fileDir === '.' ? undefined : fileDir,
-                tooltip: `${
-                  filechange.state === 'copied' || filechange.state === 'moved'
-                    ? `${filechange.oldPath} ${filechange.state === 'copied' ? '↦' : '➟'} ${
-                        filechange.newPath
-                      }`
-                    : filechange.path
-                } ${dot} ${capitalizeFirstLetter(filechange.state)}`,
-                resourceUri: Uri.file(filePath),
-                command:
-                  (filechange as FilechangeWithDiff).diff ??
-                  (filechange as FilechangeWithoutDiffButDiffViewerRegardless).current
-                    ? {
-                        command: 'radicle.openDiff',
-                        title: `Open changes`,
-                        tooltip: `Show this file's changes between its \
-before-the-Patch version and its latest version committed in the Radicle Patch`,
-                        arguments: [
-                          Uri.file(
-                            (filechange as Partial<FileChangeWithOldAndNew>).old?.oid ||
-                              (filechange as FilechangeWithoutDiffButDiffViewerRegardless)
-                                .current
-                              ? oldVersionUrl
-                              : emptyFileUrl,
-                          ),
-                          Uri.file(
-                            (filechange as Partial<FileChangeWithOldAndNew>).new?.oid ||
-                              (filechange as FilechangeWithoutDiffButDiffViewerRegardless)
-                                .current
-                              ? newVersionUrl
-                              : emptyFileUrl,
-                          ),
-                          `${filename} (${shortenHash(oldVersionCommitSha)} ⟷ ${shortenHash(
-                            newVersionCommitSha,
-                          )}) ${capitalizeFirstLetter(filechange.state)}`,
-                          { preview: true } satisfies TextDocumentShowOptions,
-                        ],
-                      }
-                    : undefined,
-              }
-
-              return filechangeTreeItem
-            },
-          }
-
-          return node
-        })
-        .sort((n1, n2) => (n1.relativeInRepoUrl < n2.relativeInRepoUrl ? -1 : 0))
-
-      return filechangeNodes.length
-        ? filechangeNodes
-        : [
-            `No changes between latest revision's base "${shortenHash(
-              latestRevision.base,
-            )}" and head "${shortenHash(latestRevision.oid)}" commits.`,
-          ]
+    const repoRoot = getRepoRoot()
+    if (repoRoot === undefined) {
+      throw new Error(`Failed to determine Git repository root.`)
     }
 
-    return undefined
+    const repo = gitExtensionApi.getRepository(Uri.file(repoRoot))
```

Comment:
Instead of getting the working copy here, it would be possible to use the bare repository. This would resolve issues of missing objects.